WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -239,16 +232,35 @@ def edit
     render :edit
   end
 
+  def clone_competition
+    @competition = Competition.new
+    cloned_competition = Competition.find(params[:id])
+    if cloned_competition
+      attributes = cloned_competition.attributes
+      # Don't clone the following attributes.
+      %w(id name cellName year month day endMonth endDay registration_open registration_close results_posted_at).each { |attribute| attributes.delete attribute }
+      @competition.assign_attributes(attributes)
+      @competition.organizers |= cloned_competition.organizers
+      @competition.delegates |= cloned_competition.delegates
+    end
+    if current_user.any_kind_of_delegate?
+      @competition.delegates |= [current_user]
+    end
+    @competition.showAtAll = false
+    @competition.isConfirmed = false
+    render :new
+  end
```

Comment:
OK. This makes sense but I think I need to keep the following in the controller since in the model it won't know about curent_user:

`    if current_user.any_kind_of_delegate?
      @clone.delegates |= [current_user]
    end`